crates/matrix-sdk/src/sync.rs

```diff
@@ -222,4 +231,11 @@ impl Client {
 
         *last_sync_time = Some(now);
     }
+
+    fn notify_sync_gap(&self, room_id: &RoomId) {
+        let mut lock = self.inner.sync_gap_broadcast_txs.lock().unwrap();
```

Comment:
Why do we need a lock for this broadcast? Do we need it for the other broadcasts? 🤔

Reply:
This isn't a single broadcast, it is one per room. The whole map containing these is behind an `RwLock`. Ultimately as I wrote in one comment, I want this to go away in favor of more general event handling anyways, because this one is not actually guaranteed to get handled before new events from the same sync are (although I think it should work out in practice).